### PR TITLE
VC Add/Invite Functionality and UI from the Space -> Community 

### PR DESCRIPTION
**Pull request by bobbykolev**
https://github.com/alkem-io/client-web/issues/7256

e2e implementation of the invite VirtualContributors functionality as per [documentation](https://github.com/alkem-io/client-web/blob/vc-store/docs/features/vcStore.md)

- [x] The Invite functionality is available in the Virtual Contributors Block under Space -> Community;
- [ ] Reusage of the functionality into (Sub)Space settings -> community; - **to be part of another branch, we can release without it!**
- [x] The VC-related parts of the Community management were extracted in a new hook (reused in the settings);
- [x] The functionality lays the foundations for re-usage for other types of Contributors (not fully);
- [x] The add/invite functionality was enhanced with disabling buttons and toast notifications, but it could require better UX regarding the loading states;
 
 Note, that a debounce was added in the fetch for VCs to guarantee proper filtering (fixing edge-case where the second request could come before the first one listing all VCs without filtering).

## Summary by CodeRabbit

- **New Features**
	- Introduced an invite dialog for inviting contributors.
	- Added functionality to display a list of contributors in a grid format.
	- Enhanced virtual contributor profile page with new sections and improved layout.

- **Bug Fixes**
	- Improved error handling in various components to ensure smooth user experience.

- **Documentation**
	- Updated interfaces to better manage contributor data and dialog interactions.

- **Refactor**
	- Streamlined privilege checks for community invite permissions.
	- Consolidated virtual contributor management into a dedicated hook.

- **Style**
	- Enhanced UI elements for better user interaction and experience.

### CODE REVIEW

**Comment by bobbykolev** [comment]
@coderabbitai review and skip any error handling suggestions.

File: src/domain/community/inviteContributors/InviteVCsDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+  } = useInviteContributors({ roleSetId, spaceId, spaceLevel });
+
+  // state
+  const [onAccount, setOnAccount] = useState<ContributorProps[]>();
+  const [inLibrary, setInLibrary] = useState<ContributorProps[]>();
+  const [openPreviewDialog, setOpenPreviewDialog] = useState(false);
+  const [openInviteDialog, setOpenInviteDialog] = useState(false);
+  const [actionButtonDisabled, setActionButtonDisabled] = useState(false);
+  const [action, setAction] = useState<'add' | 'invite'>();
+  const [selectedVirtualContributorId, setSelectedVirtualContributorId] = useState<string>('');
```
Comment (reactoholic) [inline src/domain/community/inviteContributors/InviteVCsDialog.tsx:49]: We can remove the explicit `string` generic

File: src/domain/community/inviteContributors/InviteVCsDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+        t('components.inviteContributorsDialog.successfullyAdded', {
+          contributor: t('community.virtualContributor'),
+        }),
+        'success'
+      );
+      setOpenPreviewDialog(false);
+    }
+  };
+
+  const onInviteClick = async () => {
```
Comment (reactoholic) [inline src/domain/community/inviteContributors/InviteVCsDialog.tsx:115]: Why do we use `async` here?

File: src/domain/community/inviteContributors/PreviewContributorDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  children?: ReactNode;
+}
+
+const PreviewContributorDialog = ({
+  children,
+  open = false,
+  contributor,
+  onClose,
+  actions,
+}: PreviewContributorDialogProps) => {
```
Comment (reactoholic) [inline src/domain/community/inviteContributors/PreviewContributorDialog.tsx:31]: Let's use `PropsWithChildren` and remove the explicit `children` prop in the type.

**Review by reactoholic — CHANGES_REQUESTED** [review]
PR looks good! Just small comments/questions.

File: src/domain/community/inviteContributors/InviteContributorsList.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+interface InviteContributorsListProps {
+  contributors: ContributorProps[] | undefined;
+  onCardClick?: (id: string) => void;
+}
+
+const InviteContributorsList = ({ contributors = [], onCardClick = noop }: InviteContributorsListProps) => {
+  return (
+    <GridContainer sx={{ paddingX: 0 }}>
+      {contributors.map(c => (
+        <GridItem key={c?.id}>
```
[the remark this answers is not in the document]
Comment (bobbykolev) [reply src/domain/community/inviteContributors/InviteContributorsList.tsx:16]: We can remove the ?, but I'd rather skip the filter as the VCs could get too many. In the future, we should implement a pagination.

File: src/domain/community/inviteContributors/InviteVCsDialog.tsx (hunk 2 above)
Comment (bobbykolev) [reply src/domain/community/inviteContributors/InviteVCsDialog.tsx:115]: leftover, thanks

**Pull request merged**